pybamm/models/base_model.py

```diff
@@ -422,83 +423,227 @@ def input_parameters(self):
+                else:
+                    print(f"'{submodel_name}' submodel parameters:")
+                    (
+                        max_param_name_length,
+                        max_param_type_length,
+                    ) = self._calculate_max_lengths(submodel_vars)
+
+                    table = [
+                        f"┌─{'─' * max_param_name_length}─┬─{'─' * max_param_type_length}─┐",
+                        f"│ {'Parameter':<{max_param_name_length}} │ {'Type of parameter':<{max_param_type_length}} │",
+                        f"├─{'─' * max_param_name_length}─┼─{'─' * max_param_type_length}─┤",
+                    ]
+
+                    for param, param_type in submodel_vars.values():
+                        param_name = getattr(param, "name", str(param))
+                        table.extend(
+                            self._format_table_row(
+                                param_name,
+                                param_type,
+                                max_param_name_length,
+                                max_param_type_length,
+                            )
+                        )
+                    table.extend(
+                        [
+                            f"└─{'─' * max_param_name_length}─┴─{'─' * max_param_type_length}─┘",
+                        ]
+                    )
+
+                    print("\n".join(table) + "\n")
+
+        else:
+            info = self.get_parameter_info()
+            max_param_name_length, max_param_type_length = self._calculate_max_lengths(
+                info
+            )
+
+            table = [
+                f"┌─{'─' * max_param_name_length}─┬─{'─' * max_param_type_length}─┐",
+                f"│ {'Parameter':<{max_param_name_length}} │ {'Type of parameter':<{max_param_type_length}} │",
+                f"├─{'─' * max_param_name_length}─┼─{'─' * max_param_type_length}─┤",
             ]
-            max_lines = max(len(param_name_lines), len(param_type_lines))
 
-            for i in range(max_lines):
-                param_line = param_name_lines[i] if i < len(param_name_lines) else ""
-                type_line = param_type_lines[i] if i < len(param_type_lines) else ""
-                table.append(row_format.format(param_line, type_line))
+            for param, param_type in info.values():
+                param_name = getattr(param, "name", str(param))
+                table.extend(
+                    self._format_table_row(
+                        param_name,
+                        param_type,
+                        max_param_name_length,
+                        max_param_type_length,
+                    )
+                )
 
-        for line in table:
-            print(line)
+            table.extend(
+                [
+                    f"└─{'─' * max_param_name_length}─┴─{'─' * max_param_type_length}─┘",
+                ]
+            )
+
+            print("\n".join(table) + "\n")
```

Comment:
```suggestion
            print("\n".join(table) + "\n", encoding="utf-8")
```

This should fix the windows failure.

Reply:
It's giving me an error saying `print()` doesn't have a `encoding` parameter in standard usage?

Reply:
Windows has CP1252 encoding by default. The encoding argument works when handling files, not with `print()` – we will need to process the string and convert it to UTF-8 before passing it to be printed.

Reply:
Although this should fix the failure `encoding="utf-8"` but is of no use as it prints unidentifiable characters i.e. `\xe2\x94\x8c\xe2\x94\x80\xe2\x94\x8c\xe2\x94\x80\xe2\x94\` because windows still has CP1252 encoding by default.

Setting `PYTHONIOENCODING=utf-8` in CI should actually do the work, @cringeyburger can you export this environment variable in noxfile?

Reply:
Thanks, @arjxn-py. Should we test if the characters are getting printed properly in the Jupyter notebooks where `print_parameter_info` is being called? @cringeyburger you might need to re-run the notebook to update the outputs there

Reply:
I ran `parameterization.ipynb` after updating noxfile. From what I saw, everything seems to be printed without any problems.